lms/templates/courseware/course_about.html

```diff
@@ -31,49 +36,54 @@
           type: "POST",
           /* Rant: HAD TO USE COMPLETE B/C PROMISE.DONE FOR SOME REASON DOES NOT WORK ON THIS PAGE. */
           complete: add_course_complete_handler
         })
         event.preventDefault();
       });
     % endif
-
     ## making the conditional around this entire JS block for sanity
     %if settings.FEATURES.get('RESTRICT_ENROLL_BY_REG_METHOD') and course.enrollment_domain:
       <%
-        perms_error = _('The currently logged-in user account does not have permission to enroll in this course. '
-                        'You may need to {start_logout_tag}log out{end_tag} then try the enroll button again. '
-                        'Please visit the {start_help_tag}help page{end_tag} for a possible solution.').format(
-                          start_help_tag="<a href='{url}'>".format(url=marketing_link('FAQ')), end_tag='</a>',
-                          start_logout_tag="<a href='{url}'>".format(url=reverse('logout'))
-                          )
+        perms_error = Text(
+                _('The currently logged-in user account does not have permission to enroll in this course. '
+                  'You may need to {start_logout_tag}log out{end_tag} then try the enroll button again. '
+                  'Please visit the {start_help_tag}help page{end_tag} for a possible solution.')
+                ).format(
+                  start_help_tag=HTML("<a href='{url}'>").format(url=marketing_link('FAQ')),
+                  end_tag=HTML('</a>'),
+                  start_logout_tag=HTML("<a href='{url}'>").format(url=reverse('logout'))
+                )
       %>
     $('#class_enroll_form').on('ajax:complete', function(event, xhr) {
       if(xhr.status == 200) {
-        location.href = "${reverse('dashboard')}";
+          location.href = "${reverse('dashboard')| n, js_escaped_string}";
       } else if (xhr.status == 403) {
-        location.href = "${reverse('course-specific-register', args=[course.id.to_deprecated_string()])}?course_id=${course.id | u}&enrollment_action=enroll";
-      } else if (xhr.status == 400) { //This means the user did not have permission
-        $('#register_error').html("${perms_error}").css("display", "block");
+          location.href = "${reverse('course-specific-register', args=[course.id.to_deprecated_string()])| n, js_escaped_string}?course_id=${course.id| n, js_escaped_string}&enrollment_action=enroll";
+      } else if (xhr.status == 400) {//This means the user did not have permission
+          edx.HtmlUtils.setHtml(
+                  $('#register_error'),
+                  "${perms_error}"
+          )
       } else {
-        $('#register_error').html(
-            (xhr.responseText ? xhr.responseText : "${_("An error occurred. Please try again later.")}")
+        $('#register_error').text(
+            (xhr.responseText ? xhr.responseText : "${_('An error occurred. Please try again later.')| n, js_escaped_string}")
```

Comment:
Assuming `xhr.responseText` is plain text (i.e. doesn't contain HTML tags), then this is perfect.  The browser does multiple passes over this code.  The first pass it is trying to determine what is HTML and what is script.  The `js_escaped_string` makes sure that everything looks like JavaScript and doesn't mistakenly look like HTML.  The call to `text()` is used by JavaScript to add text to the DOM, and again if anything looks like HTML, it will take care of making sure it renders as text.

Read this brief section in the docs for more understanding about [escaping in JavaScript contexts](http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/safe_templates.html#javascript-context-and-escaping).

Nit: I'd surround ` | ` with spaces on both sides.  

Also, you have two comments here, but I just see the one line.